src/Transport/CookieTransport.php

```diff
@@ -28,14 +29,40 @@ final class CookieTransport implements HttpTransportInterface
     /** @var string */
     private $basePath;
 
+    /** @var string|null */
+    private $domain;
+
+    /** @var bool */
+    private $secure;
+
+    /** @var bool */
+    private $httpOnly;
+
+    /** @var string|null */
+    private $sameSite;
+
     /**
-     * @param string $cookie
-     * @param string $basePath
+     * @param string      $cookie
+     * @param string      $basePath
+     * @param string|null $domain
+     * @param bool        $secure
+     * @param bool        $httpOnly
+     * @param string|null $sameSite
      */
-    public function __construct(string $cookie, string $basePath = '/')
-    {
+    public function __construct(
```

Comment:
We need to do something about it (in future probably). The constructor completely overlaps and duplicates the cookie constructor. Those looks like an implicit vendor lock.